gammapy/cube/fit.py

```diff
@@ -208,6 +262,65 @@ def bin_volume(self):
         de = self.energy_bin_width
         return omega * de
 
+    @property
+    def coords(self):
+        """Return evaluator coords"""
+        lon, lat = self.lon_lat
+        if self.edisp:
+            energy = self.edisp.e_reco.nodes[:, np.newaxis, np.newaxis]
+        else:
+            energy = self.energy_center
+        return {"lon": lon.value, "lat": lat.value, "energy": energy}
+
+    @property
+    def needs_update(self):
+        """Check whether the model component has drifted away from its support."""
+        if self.exposure is None:
+            update = True
+        else:
+            position = self.model.position
+            separation = self.exposure.geom.center_skydir.separation(position)
+            update = separation > UPDATE_THRESHOLD
+        return update
+
+    def update(self, exposure, psf, edisp, geom):
+        """Update MapEvaluator, based on the current position of the model component.
+
+        Parameters
+        ----------
+        exposure : `Map`
+            Exposure map.
+        psf : `PSFMap`
+            PSF map.
+        edisp : `EdispMap`
+            Edisp map.
+        geom : `MapGeom`
+            Reference geometry of the data.
+        """
+        log.debug("Updating model evaluator")
+        # TODO: lookup correct Edisp for this component
+        self.edisp = edisp
+        self.psf = psf
+
+        # TODO: lookup correct PSF for this component
+        width = np.max(psf.psf_kernel_map.geom.width) + 2 * self.model.evaluation_radius
+
+        self.exposure = exposure.cutout(position=self.model.position, width=width)
```

Comment:
Maybe I will have add a TODO comments saying that cutout will also be applied to the PSF and EDISP once IRFmaker is validated and merged!